persist/sqldb/migrate.go

```diff
@@ -215,6 +215,73 @@ func (m migrate) Exec(ctx context.Context) error {
 			ansiSQLChange(`alter table `+m.tableName+` alter column nodes type json using nodes::json`),
 		),
+		// add instanceid column to table argo_archived_workflows
+		ansiSQLChange(`alter table argo_archived_workflows add column instanceid varchar(64)`),
+		ansiSQLChange(`update argo_archived_workflows set instanceid = '' where instanceid is null`),
+		ternary(dbType == MySQL,
+			ansiSQLChange(`alter table argo_archived_workflows modify column instanceid varchar(64) not null`),
+			ansiSQLChange(`alter table argo_archived_workflows alter column instanceid set not null`),
+		),
+		// add instanceid column to table argo_archived_workflows_labels
+		ansiSQLChange(`alter table argo_archived_workflows_labels add column instanceid varchar(64)`),
+		ansiSQLChange(`update argo_archived_workflows_labels set instanceid = '' where instanceid is null`),
+		ternary(dbType == MySQL,
+			ansiSQLChange(`alter table argo_archived_workflows_labels modify column instanceid varchar(64) not null`),
+			ansiSQLChange(`alter table argo_archived_workflows_labels alter column instanceid set not null`),
+		),
+		// drop argo_archived_workflows_labels fk
+		ternary(dbType == MySQL,
+			ansiSQLChange(`alter table argo_archived_workflows_labels drop foreign key argo_archived_workflows_labels_ibfk_1`),
+			ansiSQLChange(`alter table argo_archived_workflows_labels drop constraint argo_archived_workflows_labels_clustername_uid_fkey`),
+		),
+		// drop argo_archived_workflows pk
+		ternary(dbType == MySQL,
+			ansiSQLChange(`alter table argo_archived_workflows drop primary key`),
+			ansiSQLChange(`alter table argo_archived_workflows drop constraint argo_archived_workflows_pkey`),
+		),
+		// add argo_archived_workflows pk
+		ansiSQLChange(`alter table argo_archived_workflows add primary key(clustername,instanceid,uid)`),
+		// drop argo_archived_workflows_labels pk
+		ternary(dbType == MySQL,
+			ansiSQLChange(`alter table argo_archived_workflows_labels drop primary key`),
+			ansiSQLChange(`alter table argo_archived_workflows_labels drop constraint argo_archived_workflows_labels_pkey`),
+		),
+		// add argo_archived_workflows_labels pk
+		ansiSQLChange(`alter table argo_archived_workflows_labels add primary key(clustername,instanceid,uid,name)`),
+		// add argo_archived_workflows_labels fk
+		ansiSQLChange(`alter table argo_archived_workflows_labels add constraint argo_archived_workflows_labels_fk1 foreign key (clustername,instanceid,uid) references argo_archived_workflows(clustername, instanceid, uid) on delete cascade`),
+		// drop argo_archived_workflows index
+		ternary(dbType == MySQL,
+			ansiSQLChange(`drop index argo_archived_workflows_i1 on argo_archived_workflows`),
+			ansiSQLChange(`drop index argo_archived_workflows_i1`),
+		),
+		// add argo_archived_workflows index
+		ansiSQLChange(`create index argo_archived_workflows_i1 on argo_archived_workflows (clustername,instanceid,namespace)`),
```

Comment:
do we have an index on clustername+namespace? if so, can we replace it with this one?

Reply:
yes, that one is dropped and replaced with it.